games-emulation/mame/mame-0.259.recipe

```diff
@@ -0,0 +1,231 @@
+\"vintage\" software from being lost and forgotten. This is achieved by \
+documenting the hardware and how it functions. The source code to MAME \
+serves as this documentation. The fact that the software is usable serves \
+primarily to validate the accuracy of the documentation (how else can you \
+prove that you have recreated the hardware faithfully?). Over time, MAME \
+(originally stood for Multiple Arcade Machine Emulator) absorbed the \
+sister-project MESS (Multi Emulator Super System), so MAME now documents a \
+wide variety of (mostly vintage) computers, video game consoles and \
+calculators, in addition to the arcade video games that were its initial focus."
+HOMEPAGE="https://www.mamedev.org/"
+COPYRIGHT="1997-2023 MAMEDev and contributors"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://github.com/mamedev/mame/archive/mame${portVersion/./}.tar.gz"
+CHECKSUM_SHA256="46baf431079a3373ffe8d984b3ba5d62ad5b1d5e356d1f60cf60f6ad03d4cec6"
+SOURCE_FILENAME="mame-$portVersion.tar.gz"
+SOURCE_DIR="mame-mame${portVersion/./}"
+PATCHES="mame-$portVersion.patchset"
+ADDITIONAL_FILES="
+	mame.sh
+	mame.rdef.in"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+bits=64
+optimize=3
+makejobs=$jobArgs
+commandBinDir=$binDir
+commandSuffix=$secondaryArchSuffix
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+	bits=
+	commandBinDir=$prefix/bin
+	commandSuffix=
+	makejobs=1
+	optimize=1
+fi
+
+PROVIDES="
+	mame$secondaryArchSuffix = $portVersion
+	cmd:castool$commandSuffix
+	cmd:chdman$commandSuffix
+	cmd:floptool$commandSuffix
+	cmd:imgtool$commandSuffix
+	cmd:jedutil$commandSuffix
+	cmd:ldresample$commandSuffix
+	cmd:ldverify$commandSuffix
+	cmd:mame$commandSuffix
+	cmd:nltool$commandSuffix
+	cmd:nlwav$commandSuffix
+	cmd:pngcmp$commandSuffix
+	cmd:regrep$commandSuffix
+	cmd:romcmp$commandSuffix
+	cmd:split$commandSuffix
+	cmd:src2html$commandSuffix
+	cmd:srcclean$commandSuffix
+	cmd:testkeys$commandSuffix
+	cmd:unidasm$commandSuffix
+	"
+
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libexpat$secondaryArchSuffix
+	lib:libflac$secondaryArchSuffix
+	lib:libfontconfig$secondaryArchSuffix
+	lib:libfreetype$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+	lib:libjpeg$secondaryArchSuffix
+	lib:liblua$secondaryArchSuffix >= 5.4
+	lib:libpugixml$secondaryArchSuffix
+	lib:libQt5Core$secondaryArchSuffix
+	lib:libQt5Gui$secondaryArchSuffix
+	lib:libQt5Widgets$secondaryArchSuffix
+	lib:libSDL2_2.0$secondaryArchSuffix
+	lib:libSDL2_ttf_2.0$secondaryArchSuffix
+	lib:libsqlite3$secondaryArchSuffix
+	lib:libutf8proc$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:glm
+	devel:libexpat$secondaryArchSuffix
+	devel:libflac$secondaryArchSuffix
+	devel:libfontconfig$secondaryArchSuffix
+	devel:libfreetype$secondaryArchSuffix
+	devel:libGL$secondaryArchSuffix
+	devel:libjpeg$secondaryArchSuffix
+	devel:liblua$secondaryArchSuffix >= 5.4
```

Comment:
You have to comment these references for lua also :)

Reply:
Can you share your version of the recipe, please? I just want to check that I've got it exactly right.

Reply:
This is what I'm getting now:

```
Warning: POLICY ERROR: can't find used library "liblua.so.5.4"
Warning: POLICY ERROR: can't find used library "liblua.so.5.4"
Warning: Error: packaging policy violation(s) in strict mode
cleaning 'chroot/boot' folder
keeping chroot folder /boot/home/haikuports/games-emulation/mame/work-0.259 intact for inspection
Error: Build has failed - stopping.
```

Reply:
Clean build fixed it! Installing now.